File: golang/go-demo/basic/switch.go
package main

import "fmt"

func main() {
	number := 100

	switch {
	case number < 60:
		fmt.Println("failed")
	case 60 <= number && number < 70:
		fmt.Println("pass")
	case 70 <= number && number < 80:
		fmt.Println("good")
	case 80 <= number && number < 90:
		fmt.Println("very good")
	case 90 <= number && number < 100:
		fmt.Println("excellent")
	case number == 100:
		fmt.Println("perfect")
	default:
		fmt.Println("not supported")
	}

	// fallthrough
	number = 3
	switch number {
	case 1:
		fallthrough
	case 2:
		fallthrough
	case 3:
		fallthrough
	case 4:
		fmt.Println("It's one, two, three or four")
	default:
		fmt.Println("It's other number")
	}
}

File: dotnet/CSharpExamples/CSharpExamples/Basic/Switch.cs
namespace CSharpExamples.Basic;

class SwitchShowCase
{
  [Test]
  public void SwitchStatement()
  {
    var score = 88;
    string result;

    switch (score)
    {
      case 90:
        result = "Excellent";
        break;
      case 80:
        result = "Good";
        break;
      case 70:
        result = "Average";
        break;
      default:
        result = "Needs Improvement";
        break;
    }

    Console.WriteLine($"The number is: {result}");
  }

  [Test]
  public void SwitchExpression()
  {
    var number = 2;
    var result = number switch
    {
      1 => "One",
      2 => "Two",
      3 => "Three",
      _ => "Other"
    };

    Console.WriteLine($"The number is: {result}");
  }
}
